domain/lease/state/state_test.go

```diff
@@ -1,39 +1,40 @@
-// Copyright 2022 Canonical Ltd.
+// Copyright 2023 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-package lease_test
+package state_test
 
 import (
 	"context"
 	"time"
 
 	"github.com/juju/errors"
 	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/v3"
 	gc "gopkg.in/check.v1"
 
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
 	corelease "github.com/juju/juju/core/lease"
 	"github.com/juju/juju/database/testing"
-	"github.com/juju/juju/worker/lease"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/lease/state"
 )
 
-type storeSuite struct {
+type stateSuite struct {
 	testing.ControllerSuite
 
-	store *lease.Store
+	store *state.State
 }
 
-var _ = gc.Suite(&storeSuite{})
+var _ = gc.Suite(&stateSuite{})
 
-func (s *storeSuite) SetUpTest(c *gc.C) {
+func (s *stateSuite) SetUpTest(c *gc.C) {
 	s.ControllerSuite.SetUpTest(c)
 
-	s.store = lease.NewStore(lease.StoreConfig{
-		TxnRunner: s.TxnRunner(),
-		Logger:    lease.StubLogger{},
-	})
+	s.store = state.NewState(domain.NewTxnRunnerFactory(s.getWatchableDB))
```

Comment:
You should be able to do `state.NewState(c.TxnRunner)`

Reply:
Nearly. We had to add `testing.TxnRunnerFactory` to get around the function return mismatch.

```go
s.store = state.NewState(testing.TxnRunnerFactory(s.TxnRunner()))
```